### DISCUSSION TITLE
Database > size and filtering

### DISCUSSION THREAD

**Discussion by jeca32** [jblance/mpp-solar#148] [Q&A]
the getCellData and QPIGS commands provide a lot of unnecessary data.

in order to save the sdcard, I would like to filter the filling of the database

telegraf can filter this with the function "namepass" and "namedrop" but my attempts do not work
Does anyone have the solution ?

here is an example that does not work 

/etc/telegraf/telegraf.d/mqtt-input.conf

[inputs.mqtt_consumer]]
  servers = ["tcp://127.0.0.1:1883"]
  topics = [
    "#",
  ]
  data_format = "influx"
 
namedrop = ["*voltage_cell01"]

**Comment by DMIINC** [comment]
If it is of any help I do use filter function in my jkbms.conf file.

**Reply by jeca32** [reply]
Great
can you stream a file?

**Reply by DMIINC** [reply]
> 
> 
> Great
> can you stream a file?

Can you explain stream.

**Reply by jeca32** [reply]
sorry
can you give me your file jkbms.conf

**Comment by DMIINC** [comment]
[SETUP]
### DEBUG
mqtt_broker = 192.168.1.***
max_connection_attempts = 3
records = 1
pause=2
# Uncomment one of the logging_level lines
# All messages at the uncommented level and higher will be displayed
# i.e. DEBUG gives the most output and CRITICAL gives the least
### DEBUG
#logging_level =   10
### INFO
#logging_level =  20
### WARNING
logging_level =  30
### ERROR
#logging_level =  40
### CRITICAL
#logging_level =  50

# JKBMS
[Power Wall 1]
type=jkbms
protocol=JK04
port=3c:a5:09:0a:85:79
command=getCellData
tag=CellData_1
outputs=hass_mqtt
filter=_your data choice here_

**Reply by jeca32** [reply]
I don't understand your file "filter = your data choice here"

I added your line of code but it doesn't filter anything

--------------------------

/etc/jkbms/jkbms.conf


[Bmsjeca]
type=jkbms
protocol=JK02
port=3C:A5:51:87:DD:68
command=getCellData
tag=CellData

outputs=influx2_mqtt,screen

filter = ["battery_voltage"]

---------------------------

pi@raspberrypi:~ $ /usr/local/bin/jkbms -C /etc/jkbms/jkbms.conf
Parameter                     Value           Unit
header                         55aaeb90          
record_type                   02                
record_counter                 17                
voltage_cell01                 3.282           V  
voltage_cell02                 3.285           V  
voltage_cell03                 3.283           V  
voltage_cell04                 3.283           V  
voltage_cell05                 3.283           V  
voltage_cell06                 3.283           V  
voltage_cell07                 3.281           V  
voltage_cell08                 3.282           V  
voltage_cell09                 3.283           V  
voltage_cell10                 3.282           V  
voltage_cell11                 3.283           V  
voltage_cell12                 3.283           V  
voltage_cell13                 3.282           V  
voltage_cell14                 3.283           V  
voltage_cell15                 3.282           V  
voltage_cell16                 3.281           V  
voltage_cell17                 0.0             V  
voltage_cell18                 0.0             V  
voltage_cell19                 0.0             V  
voltage_cell20                 0.0             V  
voltage_cell21                 0.0             V  
voltage_cell22                 0.0             V  
voltage_cell23                 0.0             V  
voltage_cell24                 0.0             V  
average_cell_voltage           3.282           V  
delta_cell_voltage             0.005           V  
current_balancer               2.819              
resistance_cell01             0.12           Ohm
resistance_cell02             0.122           Ohm
resistance_cell03             0.124           Ohm
resistance_cell04             0.121           Ohm
resistance_cell05             0.123           Ohm
resistance_cell06             0.122           Ohm
resistance_cell07             0.121           Ohm
resistance_cell08             0.12           Ohm
resistance_cell09             0.119           Ohm
resistance_cell10             0.119           Ohm
resistance_cell11             0.12           Ohm
resistance_cell12             0.119           Ohm
resistance_cell13             0.118           Ohm
resistance_cell14             0.12           Ohm
resistance_cell15             0.12           Ohm
resistance_cell16             0.118           Ohm
resistance_cell17             0.0             Ohm
resistance_cell18             0.0             Ohm
resistance_cell19             0.0             Ohm
resistance_cell20             0.0             Ohm
resistance_cell21             0.0             Ohm
resistance_cell22             0.0             Ohm
resistance_cell23             0.0             Ohm
resistance_cell24             0.0             Ohm
battery_voltage               52.519         V  
battery_power                 0               W  
balance_current               0.0             A  
battery_t1                     24.0           °C  
battery_t2                     23.4           °C  
mos_temp                       25.3           °C  
percent_remain                 69             %  
capacity_remain               213.903         Ah  
nominal_capacity               310.0           Ah  
cycle_count                   0                  
cycle_capacity                 4.647           Ah  
time                           25D23H8M3S        
current_charge                 0.004              
current_discharge             0.0         
       
![grafa_6](https://user-images.githubusercontent.com/89108682/131253000-0a65a632-ee78-4a53-8dfb-9c2ca0a351fc.jpg)

**Comment by DMIINC** [comment]
"your data choice" = voltage_cell01 or anything you want it to be. example below.

filter = average_cell_voltage

**Reply by jeca32** [reply]
it works with a single value

filter = voltage_cell01

but how to do with 2?
I have tested several syntax but it does not work.

filter = [delta_cell_voltage, battery_voltage]
filter = delta_cell_voltage, battery_voltage
filter = delta_cell_voltage battery_voltage
filter = delta_cell_voltage; battery_voltage
filter = "delta_cell_voltage" "battery_voltage"
filter = ["delta_cell_voltage" "battery_voltage"]

I didn t find any documentation for formatting this line

**Comment by DMIINC** [comment]
filter = average_cell_voltage | battery_voltage

**Reply by jeca32** [reply]
this syntax crashed the server
filter = delta_cell_voltage | battery_voltage

pi@raspberrypi:~ $ influx
Failed to connect to http://localhost:8086: Get http://localhost:8086/ping: dial tcp [::1]:8086: connect: connection refused
Please check your connection settings and ensure 'influxd' is running.

so, return to my old disk image of my sdcard

**Comment by jblance** [comment]
Mppsolar filter shouldn't have crashed influx

You should test each component in isolation first.
Eg stop the mppsolar service and test on the command line that the config
file works as expected (change mqttbroker to screen to see output)
Once you are happy with the output, change mqttbroker back to the correct
value and run again from command line checking telegraf and influx
If things break, undoing the change and restarting services should be enough

**Reply by jeca32** [reply]
yes i did exactly that
I think the database was corrupted because the "influx" command was not responding.

this was not a serious problem, my backup worked fine
I am still looking for a filtering solution to reduce the size of this database

**Reply by jblance** [reply]
As mentioned the filter and excl_filter options will do what you need. It'd best to work out the values using the command line e though.
Filter is used to match wanted items and excl_filter removes options
They use any valid regex

**Reply by jeca32** [reply]
if I had a working example or a tutorial, I could understand and reproduce it on my configuration.

**Comment by jblance** [comment]
Show me the output of a command line version using output=screen and tell
me what you are trying to do.
I'll try to give you an example

**Reply by jeca32** [reply]
currently this command (which is the mouth of the service) has for answer

pi@raspberrypi:~ $ /usr/local/bin/jkbms -C /etc/jkbms/jkbms.conf
Parameter Value Unit
header 55aaeb90
record_type 02
record_counter 17
voltage_cell01 3.282 V
voltage_cell02 3.285 V
voltage_cell03 3.283 V
voltage_cell04 3.283 V
voltage_cell05 3.283 V
voltage_cell06 3.283 V
voltage_cell07 3.281 V
voltage_cell08 3.282 V
voltage_cell09 3.283 V
voltage_cell10 3.282 V
voltage_cell11 3.283 V
voltage_cell12 3.283 V
voltage_cell13 3.282 V
voltage_cell14 3.283 V
voltage_cell15 3.282 V
voltage_cell16 3.281 V
voltage_cell17 0.0 V
voltage_cell18 0.0 V
voltage_cell19 0.0 V
voltage_cell20 0.0 V
voltage_cell21 0.0 V
voltage_cell22 0.0 V
voltage_cell23 0.0 V
voltage_cell24 0.0 V
average_cell_voltage 3.282 V
delta_cell_voltage 0.005 V
current_balancer 2.819
resistance_cell01 0.12 Ohm
resistance_cell02 0.122 Ohm
resistance_cell03 0.124 Ohm
resistance_cell04 0.121 Ohm
resistance_cell05 0.123 Ohm
resistance_cell06 0.122 Ohm
resistance_cell07 0.121 Ohm
resistance_cell08 0.12 Ohm
resistance_cell09 0.119 Ohm
resistance_cell10 0.119 Ohm
resistance_cell11 0.12 Ohm
resistance_cell12 0.119 Ohm
resistance_cell13 0.118 Ohm
resistance_cell14 0.12 Ohm
resistance_cell15 0.12 Ohm
resistance_cell16 0.118 Ohm
resistance_cell17 0.0 Ohm
resistance_cell18 0.0 Ohm
resistance_cell19 0.0 Ohm
resistance_cell20 0.0 Ohm
resistance_cell21 0.0 Ohm
resistance_cell22 0.0 Ohm
resistance_cell23 0.0 Ohm
resistance_cell24 0.0 Ohm
battery_voltage 52.519 V
battery_power 0 W
balance_current 0.0 A
battery_t1 24.0 °C
battery_t2 23.4 °C
mos_temp 25.3 °C
percent_remain 69 %
capacity_remain 213.903 Ah
nominal_capacity 310.0 Ah
cycle_count 0
cycle_capacity 4.647 Ah
time 25D23H8M3S
current_charge 0.004
current_discharge 0.0


and i would like to get

delta_cell_voltage 0.005 V
battery_voltage 52.519 V
battery_power 0 W
battery_t1 24.0 °C
battery_t2 23.4 °C
mos_temp 25.3 °C
percent_remain 69 %
capacity_remain 213.903 Ah
cycle_count 0
cycle_capacity 4.647 Ah
time 25D23H8M3S
current_charge 0.004
current_discharge 0.0

I would go from 68 to 13 lines and I would get database with a size reduced by 80%.
do you understand my goal? 

but I think it would be easier to filter before writing to the database rather than modifying the JK02 protocol.

**Comment by jblance** [comment]
There's a filter built in to the command, so no changes to the protocol
needed.
I'll mock up an example tomorrow

**Reply by jeca32** [reply]
Great

**Comment by jblance** [comment]
something like
`jkbms -p test -P JK02 --filter "delta|battery|temp|remain|cycle|time|current" --exclfilter "balance"`
```
MqttBroker(name='localhost', port=1883, username=None, password=None, results_topic='jkbms')
Command: getCellData - BLE Cell Data inquiry
------------------------------------------------------------
Parameter                     	Value           Unit
delta_cell_voltage            	0.025          	V   
battery_voltage               	53.185         	V   
battery_power                 	0              	W   
battery_t1                    	24.5           	°C  
battery_t2                    	21.8           	°C  
mos_temp                      	26.1           	°C  
percent_remain                	69             	%   
capacity_remain               	8.526          	Ah  
cycle_count                   	2264924160     	    
cycle_capacity                	1459617.937    	Ah  
time                          	0D0H17M21S     	    
current_charge                	0.0            	    
current_discharge             	0.0  
```

**Reply by jeca32** [reply]
thank you
I would test this tomorrow (here in France 11:44 pm)
